packages/components/src/internal/components/domainproperties/dataclasses/constants.ts

```diff
@@ -0,0 +1,16 @@
+export const DATACLASS_DOMAIN_SYSTEM_FIELDS = [
+    {
+        Name: 'Name',
+        Label: 'Name',
+        'Data Type': 'Text',
+        Required: false,
```

Comment:
for both Data Classes and Sample Types, it seems a bit odd to say that the Name isn't required. It does need to be non-null in the end, it just might not be required from the user if there is a naming pattern set for the domain. Might want to double check with Hannah on this one. I would vote we say `Required: true` for the Name fields in both.